src/interfaces/validator/validate-changes.js

```diff
@@ -73,3 +75,123 @@ export function validateChanges({incomingRecord, existingRecord, validate = true
+  debug(`Sort Aleph internal fields`);
+  const sortedNormFields = sortAlephInternalFields(normFields);
+  debugData(`OrigTags: ${normFields.map(field => field.tag)}`);
+  debugData(`SortTags: ${sortedNormFields.map(field => field.tag)}`);
+
+  //debugData(fields);
+  return new MarcRecord({leader: normLeader, fields: sortedNormFields}, {subfieldValues: false});
+
+  function emptyLeader(leader) {
+    if (!ldrIsValid(leader)) {
+      return leader;
+    }
+    // emptyLeader:
+    // - Normalize away LDR values that are 00-04 and 12-16 which are relevant only in binary MARC 21
+    // - Normalize away n/c difference in LDR/05
+    const normLeader1 = normalizeLdr05(leader);
+    const normLeader2 = emptyLdrBinaryValues(normLeader1);
+    return normLeader2;
+  }
+
+  function ldrIsValid(leader) {
+    if (!typeof leader === 'string' && !(leader instanceof String)) {
+      debug(`Non-string LDR, cannot normalize.`);
+      return false;
+    }
+    if (leader.length !== 24) {
+      debug(`Weird LDR length, cannot normalize`);
+      return false;
+    }
+    return true;
+  }
+
+  function normalizeLdr05(leader) {
+    debug(`Normalize LDR/05`);
+    const newLeader = `${leader.substring(0, 5)}${normLdr05Value(leader.substring(5, 6))}${leader.substring(6)}`;
+    debugData(`LdrOrig: ${leader}`);
+    debugData(`LdrNorm: ${newLeader}`);
+    return newLeader;
+  }
+
+  function normLdr05Value(value) {
+    if (value === 'n') {
+      return 'c';
+    }
+    return value;
+  }
+
+  function emptyLdrBinaryValues(leader) {
+    debug(`Empty LDR binary values`);
+    const newLeader = `00000${leader.substring(5, 12)}00000${leader.substring(17)}`;
+    debugData(`LdrOrig: ${leader}`);
+    debugData(`LdrNorm: ${newLeader}`);
+    return newLeader;
+  }
+
+  function emptyf008(controlfield) {
+    if (controlfield.tag !== '008' || controlfield.value.length < 6) {
+      return controlfield;
+    }
+    return {tag: controlfield.tag, value: emptyCreationDate(controlfield.value)};
+  }
+
+  function emptyCreationDate(f008value) {
+  // emptyCreationDate:
+  // Normalize f008/00-05 - In non-MARC21 imports f008 'Date entered on file' gets always the current date
+  // This propably should be configurable
+    debug(`Empty 008 creationDate.`);
+    const newValue = `000000${f008value.substring(6)}`;
+    debugData(`f008Orig: ${f008value}`);
+    debugData(`f008Norm: ${newValue}`);
+    return newValue;
+  }
+
+  // Sort Aleph internal fields (tags consisting letters) to alphabetical tag order
+  // We should have also tag-internal sort for - these are sane order in (almost) all cases for handling LOW/SID/CAT:
+  // LOW: $a contents
+  // SID: $b contents
+  // CAT: $c+$h contents
+
+  function sortAlephInternalFields(fields) {
+    const alephInternalPattern = /^[A-Z][A-Z][A-Z]$/u;
+    return [...fields].sort((a, b) => {
+      // If either of fields is and internal field do sort
```

Comment:
Can we use marc-record sort?
https://github.com/NatLibFi/marc-record-js/blob/master/src/marcFieldSort.js

Reply:
We want to isolate sorting to the Aleph-internal fields to recognise changes of field order of actual content fields as valid changes.